Source/Steam/CompatTool.cpp

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: MIT
+/*
+$info$
+tags: Bin|FEXCompatTool
+desc: Used for launching games from Steam
+$end_info$
+*/
+
+#include "PortabilityInfo.h"
+#include "Common/Config.h"
+#include "FEXCore/Utils/FileLoading.h"
+#include "FEXCore/Utils/StringUtils.h"
+#include "FEXHeaderUtils/Filesystem.h"
+
+#include <stdlib.h>
+#include <tiny-json.h>
+
+fextl::string GenerateSteamConfigTemplate(const FEX::Config::PortableInformation& PortableInfo) {
+  const auto ConfigTemplatePath = PortableInfo.InterpreterPath + "ConfigTemplate.json";
+  if (!FHU::Filesystem::Exists(ConfigTemplatePath)) {
+    return {};
+  }
+
+  fextl::string Data;
+  if (!FEXCore::FileLoading::LoadFile(Data, ConfigTemplatePath)) {
+    return {};
+  }
+
+  const char* RuntimeDir = getenv("XDG_RUNTIME_DIR");
+  if (!RuntimeDir) {
+    RuntimeDir = "/run/user/1000";
```

Comment:
1000 is a uid, not a magic number, so `geteuid()` (same as shell `id -u`) would be more appropriate. But in environments where `$XDG_RUNTIME_DIR` isn't set, `/run/user/$(id -u)` most likely isn't going to exist either, so it doesn't make a great fallback: attempts to write to it will just fail.

GLib falls back to `$XDG_CACHE_HOME` (or `~/.cache`) if `$XDG_RUNTIME_DIR` isn't set, and that seems like a reasonable "least-bad" approach to this.